ocaml/typing/typedecl.ml

```diff
@@ -1599,6 +1584,11 @@ let update_decl_jkind env dpath decl =
   let update_variant_kind cstrs rep =
     (* CR layouts: factor out duplication *)
     match cstrs, rep with
+    | _, Variant_with_null ->
+      (* CR layouts v3.5: this case only happens with [or_null_reexport].
+         Change when we allow users to write their own null constructors. *)
+      (* CR layouts v3.3: use [any_non_null]. *)
```

Comment:
I don't think we should.

Reply:
Why? I was thinking of letting people tag a single constant variant constructor with `[@@null]`, and checking that the resulting type makes sense (has no other null constructors, and is either `[@@unboxed]` with one another constructor, or a regular variant). That would be the goal of v3.5.